src/Chainweb/BlockHeaderDB/RestAPI/Client.hs

```diff
@@ -54,18 +131,56 @@ headerClient_
     . KnownChainwebVersionSymbol v
     => KnownChainIdSymbol c
     => DbKey BlockHeaderDb
-    -> ClientM (DbEntry BlockHeaderDb)
-headerClient_ = client (headerApi @v @c)
+    -> ClientM BlockHeader
+headerClient_ = headerClientContentType_ @v @c @JSON
 
 headerClient
     :: ChainwebVersion
     -> ChainId
     -> DbKey BlockHeaderDb
-    -> ClientM (DbEntry BlockHeaderDb)
-headerClient v c k = runIdentity $ do
-    SomeChainwebVersionT (_ :: Proxy v) <- return $ someChainwebVersionVal v
-    SomeChainIdT (_ :: Proxy c) <- return $ someChainIdVal c
-    return $ headerClient_ @v @c k
+    -> ClientM BlockHeader
+headerClient = headerClientJson
+
+headerClientContentType_
+    :: forall (v :: ChainwebVersionT) (c :: ChainIdT) (ct :: Type) x
+    . KnownChainwebVersionSymbol v
+    => KnownChainIdSymbol c
+    => Accept ct
+    => SupportedRespBodyContentType ct x BlockHeader
+    => (HeaderApi v c) ~ x
+    => BlockHash
+    -> ClientM BlockHeader
+headerClientContentType_ = client (Proxy @(SetRespBodyContentType ct x))
```

Comment:
Add a comment that explains why both, `SupportedRespBodyContentType` and `SetRespBodyContentType` are needed and can't be joined into a single type function or constraint.